### ISSUE TITLE
Leaves tag does not have replace: false

### ISSUE THREAD

**Issue by Lykrast** [TerraformersMC/Terrestria#199]
Minecraft - 1.16.4
Terrestria - 2.14

The override for the `leaves` block tag does not have `"replace": false`, unlike other mods that add trees or even Traverse, and also has the minecraft leaves in there.

~~Also the Redwood leaves from Terrestria aren't affected by [this leaf decay mod](https://www.curseforge.com/minecraft/mc-mods/leaf-decay), while the Sakura one and the ones from Traverse are.~~ It's likely unrelated to this issue but I mentionned it cause I initially thought it could be linked.

**Comment by Prospector** [comment]
Replace is false by default, so there's no difference there. As for the inclusion of the minecraft leaves, that is because of FabricMC/Fabric#66, a bug which means vanilla tag jsons are fully overridden by mod jsons in dev environments. Do you know of any issues associated with including vanilla entries? I was not aware of any.

**Comment by Lykrast** [comment]
I am not aware of any either, I just looked into the tags after seeing the leaf decay mod not work cause I thought it might be what causes the issue (no since some Terrestria leaves work and also the leaf decay's source does not use the tag at all), and it seemed wrong to have the minecraft ones (and no replace false) cause other mods didn't do that.

**Comment by coderbot16** [comment]
The issue with redwood leaves not working with the leaf decay mod is because the leaf decay mod uses an `instanceof` check with LeavesBlock instead of checking `BlockTags.LEAVES`: https://github.com/Draylar/leaf-decay/blob/1.16/src/main/java/draylar/leafdecay/scheduler/LeafBreakHandler.java#L51

ExtendedLeavesBlock, used by Redwood and Hemlock leaves, doesn't subclass LeavesBlock because it uses a distance property that supports leaves being farther from trees than vanilla. Not extending LeavesBlock actually prevents a crash here because otherwise the mod would have tried to retrieve the DISTANCE property which isn't present on redwood and hemlock leaves.

**Comment by saltyseadoggo** [comment]
@Prospector I cannot think of any issue with adding the vanilla leaves to the tag again, with one exception: if a recipe were to call on the #leaves item tag, would it show the vanilla leaves twice each in the recipe book as it scrolls through the available leaves?

**Comment by coderbot16** [comment]
I've reported the leaf decay issue to Draylar at https://github.com/Draylar/leaf-decay/issues/11

**Comment by coderbot16** [comment]
> I cannot think of any issue with adding the vanilla leaves to the tag again, with one exception: if a recipe were to call on the #leaves item tag, would it show the vanilla leaves twice each in the recipe book as it scrolls through the available leaves?

No, tags use a Set object to ensure that a given block can show up in the tag only once.

**Comment by coderbot16** [comment]
Closing this in favor of https://github.com/TerraformersMC/Terrestria/issues/221 and https://github.com/Draylar/leaf-decay/issues/11
